iothub/device/src/IotHubConnectionStringBuilder.cs

```diff
@@ -294,9 +295,14 @@ private void SetIotHubName()
         {
             IotHubName = GetIotHubName(HostName);
 
+            // We expect the hostname to be of the format "acme.azure-devices.net", in which case the IotHubName is "acme".
+            // For transparent gateway scenarios, we can simplify the input credentials to only specify the gateway device hostname,
+            // instead of having to specify both the IoT Hub hostname and the gateway device hostname.
+            // In this case, the hostname will be of the format "myGatewayDevice", and will not have ".azure-devices.net" suffix.
             if (IotHubName.IsNullOrWhiteSpace())
             {
-                throw new FormatException("Missing IOT hub name");
+                Logging.Info(this, $"Connecting to a gateway device with hostname=[{HostName}]");
+                IotHubName = HostName;
```

Comment:
There is no additional validation added here to test if this is indeed a gateway hostname or not since that is out of scope for the sdk.
In case an invalid host is specified here, service will throw the appropriate error.